lib/src/collection/treeset.dart

```diff
@@ -690,7 +688,7 @@ class AvlTreeSet<V> extends TreeSet<V> {
     AvlNode<V> x = _root;
     int compare = 0;
     while (x != null) {
-      compare = comparator(element, x.object);
+      compare = comparator(element as V, x.object);
```

Comment:
Similar situation here. It seems everywhere we take `Object` we first should check

```dart
if (element is! V) {
  // return false, or otherwise ignore the element
}
```